integr8ly/cloud-resource-operator
# Deploy CRO resources in the cluster VPC

## Overview

Jira: https://issues.redhat.com/browse/INTLY-4954

## Verification
- Run `make cluster/prepare`
- Run `make cluster/seed/managed/postgres`
- Run `make cluster/seed/managed/redis`
- Run `make run`
- To make it easy to verify, create two workshop resources
```
apiVersion: integreatly.org/v1alpha1
kind: Postgres
metadata:
  name: example-postgres-local
  labels:
    productName: productName
spec:
  secretRef:
    name: example-postgres-sec-local
  tier: development
  type: workshop
```
```
apiVersion: integreatly.org/v1alpha1
kind: Redis
metadata:
  name: example-redis-local
  labels:
    productName: productName
spec:
  secretRef:
    name: example-redis-sec-local
  tier: development
  type: workshop
```
- Once both `managed` resources have completed, log into AWS console
- Navigate to `elasticache` expand the drop down menu on the newly created cache
- Ensure the `elasticache` `subnet group` name begins with your `clusterID`
- Make note of the `security group`
- Navigate to `rds` and click on the newly created resource
- Ensure the `subnets` are `private subnets`
- Ensure the `security group` matches that of the `elasticache` resource
- Make note of the `vpc`
- Navigate to `ec2` and find a node of your cluster and click on it`
- Check that the `vpc` matches that of the `rds` `vpc`
- Open your `Openshift console` and navigate to the secrets created by `CRO` with the connection details of the managed resources and make notes of these details
- Navigate to the `Postgres` pod running in cluster deployed by `CRO` workshop
- Using the pod terminal run the following command
```
psql --host=<<host from postgres sec>> --port=5432 --username=postgres --password --dbname=postgres
```
- When prompted enter the password from the postgres sec. 
- Ensure connection can be made to the postgres database
- Navigate to `Redis` pod running in cluster deployed by `CRO` workshop
- Using the pod terminal run the following command
```
redis-cli -h <<host from redis sec>> -p <<port from redis sec>>
```
- Ensure connection can be made to redis cache

## Follow on Work
- Clean up Subnet Group and Security Group 
- Elasticache does not support tagging of a Subnet Group
